subsys/net/lib/lwm2m/lwm2m_engine.h

```diff
@@ -30,63 +30,286 @@
+ *
+ * @param[in] obj_inst_id Object instance id of the server object instance.
+ * @return int32_t pmax value.
+ */
 int32_t lwm2m_server_get_pmax(uint16_t obj_inst_id);
+
+/**
+ * @brief Returns the Short Server ID of the server object instance with
+ * object instance id given by @p obj_inst_id.
+ *
+ * @param[in] obj_inst_id Object instance id of the server object.
+ * @return SSID or negative in case not found.
+ */
 int lwm2m_server_get_ssid(uint16_t obj_inst_id);
+
+/**
+ * @brief Returns the object instance id of the server having ssid given by @p short_id.
+ *
+ * @param[in] short_id ssid of the server object.
+ * @return Object instance id or negative in case not found.
+ */
 int lwm2m_server_short_id_to_inst(uint16_t short_id);
 
 #if defined(CONFIG_LWM2M_SERVER_OBJECT_VERSION_1_1)
 bool lwm2m_server_get_mute_send(uint16_t obj_inst_id);
 #endif
 
 #if defined(CONFIG_LWM2M_FIRMWARE_UPDATE_OBJ_SUPPORT)
+/**
+ * @brief Sets the update state (as specified in  LWM2M SPEC E.6 regarding the firmware update)
+ * of the Firmware object instance given by @p obj_inst_id to @p state.
+ * (i.e Sets the value of resource 5/ @p obj_inst_id /3)
+ *
+ * @param[in] obj_inst_id Object instance id of the firmware object instance.
+ * @param[in] state (STATE_IDLE through STATE_UPDATING)
+ */
 void lwm2m_firmware_set_update_state_inst(uint16_t obj_inst_id, uint8_t state);
+
+/**
+ * @brief Sets the result state (as specified in  LWM2M SPEC E.6 regarding the firmware update)
+ * of the Firmware object instance given by @p obj_inst_id to @p result.
+ * (i.e Sets the value of resource 5/ @p obj_inst_id /5)
+ *
+ * @param[in] obj_inst_id Object instance id of the firmware object instance.
+ * @param[in] result (RESULT_DEFAULT through RESULT_UNSUP_PROTO)
+ */
 void lwm2m_firmware_set_update_result_inst(uint16_t obj_inst_id, uint8_t result);
+
+/**
+ * @brief Equivalent to lwm2m_firmware_set_update_state_inst(0, state).
+ *
+ * @param[in] state
+ */
 void lwm2m_firmware_set_update_state(uint8_t state);
+
+/**
+ * @brief Equivalent to lwm2m_firmware_set_update_result_inst(0, result).
+ *
+ * @param[in] result
+ */
 void lwm2m_firmware_set_update_result(uint8_t result);
+
+/**
+ * @brief Returns the update state (as specified in  LWM2M SPEC E.6 regarding the firmware update)
+ * of the Firmware object instance given by @p obj_inst_id.
+ * (i.e Gets the value of resource 5/ @p obj_inst_id /3)
+ * @param[in] obj_inst_id
+ * @return (STATE_IDLE through STATE_UPDATING)
+ */
 uint8_t lwm2m_firmware_get_update_state_inst(uint16_t obj_inst_id);
+
+/**
+ * @brief Equivalent to lwm2m_firmware_get_update_state_inst(0).
+ *
+ * @return uint8_t
+ */
 uint8_t lwm2m_firmware_get_update_state(void);
+
+/**
+ * @brief Returns the result state (as specified in  LWM2M SPEC E.6 regarding the firmware update)
+ * of the Firmware object instance given by @p obj_inst_id.
+ * (i.e Gets the value of resource 5/ @p obj_inst_id /5)
+ * @param[in] obj_inst_id
```

Comment:
Description is missing. There are a few more places like this, I hope you see what I mean, I don't want to spam too much.